### ISSUE TITLE
Replace Socket.IO with a lighter-weight alternative

### ISSUE THREAD

**Issue by jniles** [huytd/agar.io-clone#188]
This has been mentioned before, but I'd like to get a public poll.  Socket.IO is a bit overkill for the current project scope.  My vote is for the [ws](https://einaros.github.io/ws/).  In particular, look at the comparisons between frameworks [here](https://einaros.github.io/ws/benchmarks.html) and [here](https://einaros.github.io/ws/servers/index.html).

Has anyone had experience with other frameworks more suitable to agar.io-clone?  Can we decide on a new alternative to put together some migration code?

**Comment by JoeMattie** [comment]
It needs to be done, no question about it.  This is probably the most pressing issue preventing a game like this from scaling (with implementation of quadtrees or similar partitioning of players and entities being a close second).

I found some good info on the subject here:

http://buildnewgames.com/optimizing-websockets-bandwidth/

**Comment by owlg** [comment]
I agree that WS is a better option. It solves many of the laggy issues!

**Comment by keonkim** [comment]
how about sock.js?

**Comment by jniles** [comment]
@keonkim , did you give sock.js a try?  My concern is that sockjs doesn't use raw websockets and if we are going to the trouble of re-writing, we should be the lightest and best possible.

**Comment by Buskcoin** [comment]
The Agar.io server implementation [Ogar](https://github.com/forairan/Ogar) is using  [ws](https://github.com/websockets/ws). Would that work well enough?

**Comment by IgorAntun** [comment]
I think so. `ws` is great.

**Comment by keonkim** [comment]
@jniles I haven't touched it yet, I also think going for `ws` is better choice though.

**Comment by jniles** [comment]
Sweet.  I propose that refactoring efforts focus on making sure we have testable components and test cases for those components, then we can swap out bits and see how it goes.

**Comment by abalabahaha** [comment]
I seem to have gotten a socket.io -> einaros/ws transition on one of my branches, still needs a lot of testing though.

On a side note, every website that uses the current tagline would be wrong if we were to switch off socket.io.
